From 1aec809cd0847a70699ec0db7f48f7bbdfe4a50c Mon Sep 17 00:00:00 2001
From: Punit Agrawal <punit.agrawal@arm.com>
Date: Tue, 8 Oct 2013 12:20:13 +0100
Subject: [PATCH] arm: cci: Make the PMU valid across all CPUs

The CCI PMU is not a CPU PMU. As such the CCI PMU events can be
initiate from any proocessor. Set the valid_cpus mask to indicate
this.

Signed-off-by: Punit Agrawal <punit.agrawal@arm.com>
Signed-off-by: Jon Medhurst <tixy@linaro.org>
(cherry picked from commit 9f496a8ab96ac3c94e1bd283e1816ae901355979)
Signed-off-by: Alex Shi <alex.shi@linaro.org>
---
 drivers/bus/arm-cci.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bus/arm-cci.c b/drivers/bus/arm-cci.c
index 8965fb5..bc8e1af 100644
--- a/drivers/bus/arm-cci.c
+++ b/drivers/bus/arm-cci.c
@@ -545,6 +545,7 @@ static int cci_pmu_init(struct arm_pmu *cci_pmu, struct platform_device *pdev)
 
 	cci_pmu->plat_device = pdev;
 	cci_pmu->num_events = pmu_get_max_counters();
+	cpumask_setall(&cci_pmu->valid_cpus);
 
 	return armpmu_register(cci_pmu, -1);
 }
-- 
1.8.0

